aws_lambda_builders/workflows/go_modules/validator.py

```diff
@@ -0,0 +1,68 @@
+"""
+Go Runtime Validation
+"""
+
+import logging
+import os
+import subprocess
+
+from aws_lambda_builders.exceptions import MisMatchRuntimeError
+
+LOG = logging.getLogger(__name__)
+
+
+class GoRuntimeValidator(object):
+
+    LANGUAGE = "go"
+    SUPPORTED_RUNTIMES = {
+        "go1.x"
+    }
+
+    def __init__(self, runtime):
+        self.runtime = runtime
+        self._valid_runtime_path = None
+
+    def has_runtime(self):
+        """
+        Checks if the runtime is supported.
+        :param string runtime: Runtime to check
+        :return bool: True, if the runtime is supported.
+        """
+        return self.runtime in self.SUPPORTED_RUNTIMES
+
+    def validate(self, runtime_path):
+        """
+        Checks if the language supplied matches the required lambda runtime
+        :param string runtime_path: runtime to check eg: /usr/bin/go
+        :raises MisMatchRuntimeError: Version mismatch of the language vs the required runtime
+        """
+        if not self.has_runtime():
+            LOG.warning("'%s' runtime is not "
+                        "a supported runtime", self.runtime)
+            return None
+
+        expected_major_version = int(self.runtime.replace(self.LANGUAGE, "").split('.')[0])
+        min_expected_minor_version = 11 if expected_major_version == 1 else 0
+
+        p = subprocess.Popen([runtime_path, "version"],
+                             cwd=os.getcwd(),
+                             stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        out, _ = p.communicate()
+
+        if p.returncode == 0:
+            out_parts = out.decode().split()
+            if len(out_parts) >= 3:
+                version_parts = [int(x) for x in out_parts[2].replace(self.LANGUAGE, "").split('.')]
+                if len(version_parts) == 3:
+                    if version_parts[0] == expected_major_version and version_parts[1] >= min_expected_minor_version:
```

Comment:
It would be nice to have sem version comparison somewhere. I am ok with this for now, I think we will see more of this and either should look at using a library or abstract this into a method any workflow could use.

Reply:
I agree, I can build a small library function that can be shared among workflows and put it in https://github.com/awslabs/aws-lambda-builders/blob/develop/aws_lambda_builders/utils.py.

Reply:
I don't want to block the PR on this. We need to do this "provide utils easily to workflows" in general. To this point, we have copied common code across the workflows. This was to keep workflows completely independent but (as you can probably tell with OSUtils being everywhere), we need a better way to do it. I lump this into there. :)